flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/services/HeadlessClusterIPService.java

```diff
@@ -50,6 +50,7 @@ public Service buildUpInternalService(
                 .withNewMetadata()
                 .withName(serviceName)
                 .withLabels(kubernetesJobManagerParameters.getCommonLabels())
+                .withAnnotations(kubernetesJobManagerParameters.getInternalServiceAnnotations())
```

Comment:
This change is not tested. I cannot see an existing test for this method, and am not sure why that is missing. Can we add one here: `InternalServiceDecoratorTest` ?

Reply:
Done.